front/components/assistant/conversation/input_bar/editor/useCustomEditor.tsx

```diff
@@ -60,9 +60,13 @@ const useEditorService = (editor: Editor | null) => {
     return {
       // Insert mention helper function
       insertMention: ({ id, label }: { id: string; label: string }) => {
+        const shouldAddSpaceBeforeMention =
+          !editor?.isEmpty &&
+          editor?.getText()[editor?.getText().length - 1] !== " ";
```

Comment:
I'd have trim and always added a space before adding the mention.

Reply:
style fight :)

Reply:
if somebody does a carriage return, and goes to pick an assistant, then you don't want to add a space